app.js

```diff
@@ -47,36 +73,45 @@ app.use(require("connect-flash")());
 //   })
 // );
 
-// secret word handling
-let secretWord = "syzygy";
-app.get("/secretWord", (req, res) => {
-  // res.render("secretWord", { secretWord });
-  if (!req.session.secretWord) {
-    req.session.secretWord = "syzygy";
-  }
-  res.locals.info = req.flash("info");
-  res.locals.errors = req.flash("error");
-  res.render("secretWord", { secretWord: req.session.secretWord });
-});
+
+//replace the app.get and app.post statements for the "/secretWord" 
+// Turning on protection is simple. You add the authentication middleware "auth" to the route 
+
+app.use("/secretWord", auth, secretWordRouter);
+
+
+// // secret word handling
+// let secretWord = "syzygy";
+// app.get("/secretWord", (req, res) => {
+//   // res.render("secretWord", { secretWord });
+//   if (!req.session.secretWord) {
+//     req.session.secretWord = "syzygy";
+//   }
+//   res.locals.info = req.flash("info");
+//   res.locals.errors = req.flash("error");
+//   res.render("secretWord", { secretWord: req.session.secretWord });
+// });
+// // app.post("/secretWord", (req, res) => {
+// //   secretWord = req.body.secretWord;
+// //   res.redirect("/secretWord");
+// // });
 // app.post("/secretWord", (req, res) => {
-//   secretWord = req.body.secretWord;
+//   if (req.body.secretWord.toUpperCase()[0] == "P") {
+//     req.flash("error", "That word won't work!");
+//     req.flash("error", "You can't use words that start with p.");
+//   } else {
+//     req.session.secretWord = req.body.secretWord;
+//     req.flash("info", "The secret word was changed.");
+//   }
+//   // res.render("secretWord", {
+//   //   secretWord,
+//   //   errors: flash("errors"),
+//   //   info: flash("info"),
+//   // });
 //   res.redirect("/secretWord");
 // });
-app.post("/secretWord", (req, res) => {
-  if (req.body.secretWord.toUpperCase()[0] == "P") {
-    req.flash("error", "That word won't work!");
-    req.flash("error", "You can't use words that start with p.");
-  } else {
-    req.session.secretWord = req.body.secretWord;
-    req.flash("info", "The secret word was changed.");
-  }
-  // res.render("secretWord", {
-  //   secretWord,
-  //   errors: flash("errors"),
-  //   info: flash("info"),
-  // });
-  res.redirect("/secretWord");
-});
+
+//end comment 05/20/2024
```

Comment:
Nice thing about Git is it keeps a timestamp record of changes for you. Dating comments can still be helpful though if you want to keep track of concepts like here :)

Reply:
Okay - thank you for letting me know that :)